lib/bolt/transport/ssh/connection.rb

```diff
@@ -309,6 +309,13 @@ def write_remote_executable(dir, file, filename = nil)
           remote_path
         end
 
+        def write_executable_from_content(dest, content, filename = nil)
```

Comment:
filename is required.  Letting it be `nil` when that's going to make us try to upload content to a directory name seems bad.